packages/components/src/components/common/table/hooks/hooks.tsx

```diff
@@ -119,3 +119,143 @@ export const useMultiSelectRow = <TData,>(table: Table<TData>) => {
+    event.stopPropagation();
+
+    switch (event.key) {
+      case 'ArrowUp':
+      case 'ArrowDown': {
+        handleArrowKeyUpDown(event);
+        break;
+      }
+      case 'ArrowRight':
+        toggleExpand(table.getSelectedRowModel().flatRows, true, options?.lazyLoadChildren);
+        break;
+      case 'ArrowLeft':
+        toggleExpand(table.getSelectedRowModel().flatRows, false, options?.lazyLoadChildren);
+        break;
+      case 'Tab':
+        table.resetRowSelection();
+        break;
+      case 'Enter':
+        handleEnterKey(table.getSelectedRowModel().flatRows, onEnterAction);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleArrowKeyUpDown = (event: React.KeyboardEvent<HTMLTableElement>) => {
+    const { multiSelect = false, reorder } = options || {};
+    const allRows = table.getRowModel().rows;
+    const selectedRows = table.getSelectedRowModel().flatRows;
+    const direction = event.key === 'ArrowUp' ? -1 : 1;
+
+    const newReorderIndex = calculateNewReorderIndex(
+      direction,
+      allRows.indexOf(selectedRows[0]),
+      allRows.indexOf(selectedRows[selectedRows.length - 1]),
+      selectedRows.length,
+      allRows.length
+    );
+    const newSelectIndex = calculateNewSelectIndex(direction, newReorderIndex, allRows.length);
+    if (reorder?.updateOrder && reorder.getRowId && event.altKey) {
+      const moveIndexes = selectedRows.map(row => row.index);
+      const moveIds = selectedRows.map(row => reorder.getRowId!(row.original));
+      reorder.updateOrder(moveIndexes, newReorderIndex);
+      resetAndSetRowSelection(table, data, moveIds, reorder.getRowId);
+      setRootIndex(newReorderIndex);
+    } else if (multiSelect && event.shiftKey) {
+      toggleRowWithShift(direction, newSelectIndex, allRows, selectedRows);
+    } else {
+      table.resetRowSelection();
+      allRows[newReorderIndex].toggleSelected();
+      setRootIndex(newReorderIndex);
+    }
+  };
+
+  const toggleRowWithShift = <TData,>(
+    direction: -1 | 1,
+    newIndex: number | undefined,
+    allRows: Array<Row<TData>>,
+    selectedRows: Array<Row<TData>>
+  ): void => {
+    if (newIndex === undefined) {
+      return;
+    }
+    if (rootIndex === undefined && selectedRows.length === 1) {
+      setRootIndex(selectedRows[0].index);
+    }
+    if (direction === 1) {
+      if (rootIndex === allRows.indexOf(selectedRows[0]) || selectedRows.length === 1) {
+        allRows[newIndex].toggleSelected();
+      } else {
+        selectedRows[0].toggleSelected();
+      }
+    } else {
+      if (rootIndex === allRows.indexOf(selectedRows[selectedRows.length - 1]) || selectedRows.length === 1) {
+        allRows[newIndex].toggleSelected();
+      } else {
+        selectedRows[selectedRows.length - 1].toggleSelected();
+      }
+    }
+  };
+
+  return { handleKeyDown };
+};
+
+const handleEnterKey = <TData,>(selectedRows: Array<Row<TData>>, onEnterAction?: (row: Row<TData>) => void) => {
```

Comment:
Yes, that’s correct. For now, I would also reduce it to a single row since the onEnterAction only works with one row anyway. I would expand it to support multiple rows only once we have a clear use case for it. Additionally, I can do without the handleEnterKey method and directly call the onEnterAction function, as I’m expecting only a single row in any case.
![grafik](https://github.com/user-attachments/assets/03b5b67f-396d-4e7b-aded-b8db62841827)